### PR TITLE
Rotated hexagons

### PR DESCRIPTION
**Pull request by mcnees**
Rewrote the hexagon and triangle pattern functions to include a rotation parameter. Removed the iso pattern code, changed the iso option to call triangles[rotate=90], added a hexup option that calls hexagons[rotate=90].

### CODE REVIEW

File: gridpapers.sty [hunk 1]
```diff
@@ -195,26 +195,33 @@
...
+	    \pgfpathlineto{\pgfpoint{0.5*\GP@patternsize}{0mm}}
+	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{-0.866025*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{2*\GP@patternsize}{-0.866025*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{2.5*\GP@patternsize}{0mm}}
+	    \pgfpathlineto{\pgfpoint{3*\GP@patternsize}{0mm}}
+	    \pgfpathmoveto{\pgfpoint{0.5*\GP@patternsize}{0mm}}
+	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{0.866025*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{2*\GP@patternsize}{0.866025*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{2.5*\GP@patternsize}{0mm}}
+	    \pgfusepath{stroke}
```
Comment (duetosymmetry) [inline gridpapers.sty]: Mixing tabs and spaces, can we try to avoid tabs?

File: gridpapers.sty [hunk 2]
```diff
@@ -195,26 +195,33 @@
...
+\tikzdeclarepattern{
+  name=hexagons,
+  type=uncolored,
+  bounding box={(0,0) and (3*\GP@patternsize,0.866025*2*\GP@patternsize)},
+  tile size={(3*\GP@patternsize,0.866025*2*\GP@patternsize)},
+  parameters={\tikzhexrotate},
+  tile transformation={rotate=\tikzhexrotate},
+  defaults={
+    rotate/.store in=\tikzhexrotate,rotate=0,
+    % tile size/.store in=\tikztilesize,tile size=10pt,
```
Comment (duetosymmetry) [inline gridpapers.sty]: Do we need to keep this commented line around?

File: graph paper.tex [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
-\usepackage[pattern=std]{gridpapers}
+\usepackage[pattern=hexup]{gridpapers}
 
-\begin{document}
+\begin{document}  
 \thispagestyle{empty}
-~
-\end{document}
+~ 
+\end{document}
```
Comment (duetosymmetry) [inline graph paper.tex]: Does this file need to change for this commit?

File: .graph paper.watcher_pid [hunk 4]
```diff
@@ -0,0 +1 @@
+89775
```
Comment (duetosymmetry) [inline .graph paper.watcher_pid]: Please remove this file, and the files `.graph paper.lb`, `.DS_Store` from the commit

File: graph paper.tex [hunk 5]
```diff
@@ -1,6 +1,6 @@
 \documentclass{article}
 
-\usepackage[pattern=hexup]{gridpapers}
+\usepackage[pattern=iso,minorcolor=cornflower!30]{gridpapers}
 
 \begin{document}  
 \thispagestyle{empty}
```
Comment (duetosymmetry) [inline graph paper.tex]: Does this file need to be touched for this commit?

File: gridpapers.sty [hunk 6]
```diff
@@ -232,53 +232,46 @@
...
+	 	% 	    \pgfpathlineto{\pgfpoint{0.5*\GP@patternsize}{0mm}}
+	 	% 	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{-0.866025*\GP@patternsize}}
+	 	% 	    \pgfpathlineto{\pgfpoint{2*\GP@patternsize}{-0.866025*\GP@patternsize}}
+	 	% 	    \pgfpathlineto{\pgfpoint{2.5*\GP@patternsize}{0mm}}
+	 	% 	    \pgfpathlineto{\pgfpoint{3*\GP@patternsize}{0mm}}
+	 	% 	    \pgfpathmoveto{\pgfpoint{0.5*\GP@patternsize}{0mm}}
+	 	% 	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{0.866025*\GP@patternsize}}
+	 	% 	    \pgfpathlineto{\pgfpoint{2*\GP@patternsize}{0.866025*\GP@patternsize}}
+	 	% 	    \pgfpathlineto{\pgfpoint{2.5*\GP@patternsize}{0mm}}
+	 	% 	    \pgfusepath{stroke}
```
Comment (duetosymmetry) [inline gridpapers.sty]: Pretty sure we can delete these commented lines

File: gridpapers.sty [hunk 7]
```diff
@@ -232,53 +232,46 @@
...
+	      \pgfsetlinewidth{0.6pt}
+	    \pgfpathmoveto{\pgfpoint{0mm}{0mm}}
+	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{2*0.8660254*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{0mm}{2*0.8660254*\GP@patternsize}}
+	      \pgfpathmoveto{\pgfpoint{0mm}{0.8660254*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{0.8660254*\GP@patternsize}}
+	      \pgfpathmoveto{\pgfpoint{0mm}{2*0.8660254*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{0mm}}
+	    \pgfpathlineto{\pgfpoint{0mm}{0mm}}
+	      \pgfusepath{stroke}
```
Comment (duetosymmetry) [inline gridpapers.sty]: Mixing tabs and spaces, let's try to avoid tabs

**Review by duetosymmetry — CHANGES_REQUESTED** [review]
Looking nice! Besides the comments below, we should also:
- Update the README with the new pattern
- Update the `.dtx` with both the new code, and documentation of new pattern
- Bump the version number and add a `\changes{}` entry in the `.dtx`
These can be done in a separate documentation commit.

Do you know how to amend/rebase your commits?

File: gridpapers.sty (hunk 1 above)
Comment (mcnees) [reply gridpapers.sty]: Yes!

File: gridpapers.sty (hunk 2 above)
Comment (mcnees) [reply gridpapers.sty]: No!

File: graph paper.tex (hunk 3 above)
Comment (mcnees) [reply graph paper.tex]: No!

File: .graph paper.watcher_pid (hunk 4 above)
Comment (mcnees) [reply .graph paper.watcher_pid]: Will do

File: graph paper.tex (hunk 5 above)
Comment (mcnees) [reply graph paper.tex]: No, I’m being super sloppy

File: gridpapers.sty (hunk 6 above)
Comment (mcnees) [reply gridpapers.sty]: Yes, I forgot to save the file in my editor before pushing the changes

File: gridpapers.sty (hunk 7 above)
Comment (mcnees) [reply gridpapers.sty]: Got it

**Comment by mcnees** [comment]
Follow-up: I do not know how to amend/rebase my commits.

**Comment by duetosymmetry** [comment]
Ok, go ahead and add more commits onto the end of the history here. When we merge we will use the 'squash and merge' option and it will squash everything down to one commit. If you undo the unwanted changes, they will get squashed down to nothing.

**Comment by mcnees** [comment]
Okay, I think I got it all.

**Comment by duetosymmetry** [comment]
Almost there... note that if you use the `.ins` and `.dtx` files to generate the `.sty` file, the generated file does not match what you have in the commit. That is, try doing the following:

    pdflatex gridpapers.ins
(and say 'y' when it prompts you to overwrite). Then with a git diff you will see that the generated `.sty` file differs.

Other small changes: the .DS_Store file is still in the commit, and there is a diff between master and this branch in the file 'graph paper.tex'. That latter diff is that there used to be a newline terminating the file, but the newline was deleted.

**Comment by mcnees** [comment]
Okay, the .sty file in the commit is the one generated from the .ins and .dtx files, and I think everything else has been fixed?

**Comment by duetosymmetry** [comment]
Looks almost ready to me! The last thing is that when we compile the documentation from the .dtx file, we get a warning about the checksum not passing,

    ! Package doc Error: Checksum not passed (432<>405).

Their simplistic approach to a checksum is just counting number of backslashes; we have to update this number whenever we update the .dtx. Just change [the 432 on line 53](https://github.com/mcnees/LaTeX-Graph-Paper/blob/68762a7e3dad067a23aa9d66f0e1e813632fbff3/gridpapers.dtx#L53) to a 405.

If you want to feel really virtuous, there is also trailing whitespace ending lines 76, 126, 188, 378, and 678; this gets stripped when generating the .sty.

**Comment by mcnees** [comment]
Fixed the checksum and trailing whitespace, and also replaced

    [2021/03/27 v1.0.2 Graph paper backgrounds]

after \ProvidesPackage{gridpapers}.

You can probably tell I've never used github for anything more than stashing files.

**Comment by mcnees** [comment]
Whoops, that changes the line in the .sty file, so I had to update that too. Jfc.

**Comment by duetosymmetry** [comment]
> Fixed the checksum and trailing whitespace, and also replaced
> 
> ```
> [2021/03/27 v1.0.2 Graph paper backgrounds]
> ```
> 
> after \ProvidesPackage{gridpapers}.
> 

Ah, good catch! I missed that. LGTM, going to hit squash and merge! Yippee!

**Review by duetosymmetry — APPROVED** [review]

**Pull request merged**